els-front/src/components/Dashboard.js

```diff
@@ -1,8 +1,45 @@
-import React from "react";
+import React, { useState } from "react";
+import { Link } from "react-router-dom";
+import Activities from "./Activities";
+import useUser from "../hooks/useUser";
+import LearnedWords from "./LearnedWords";
 
 const Dashboard = () => {
+    const { user } = useUser();
+    const [activeTab, setActiveTab] = useState('learned_words');
+
+    const renderActiveTab = () => {
+        if(activeTab === 'learned_words')
+            return <LearnedWords user={user}/>
+        else if (activeTab === 'activities')
+            return <Activities user={user}/>
+
+        return null;
+    }
+
+    if(!user) return null;
```

Comment:
I noticed that you are applying this on all components, so apply DRY. Dont Repeat Yourself. 

You can just apply this checking in useUser instead or make a useContext that wraps the whole authenticated route/app that checks user then apply necessary logic.

Or I think you already applied it in protected route. Is that not working?

Reply:
Its working sir the purpose of it is for the error that shows when the component rendered first before the API call was finished in the useUser hook

Reply:
Is it okay to just set the default values of the user in useUser hook ?

Reply:
I understand now. Maybe you can use loading screen instead of null.